File: trunk/FireflowSuite/engine/src/org/fireflow/engine/calendar/ICalendarService.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package org.fireflow.engine.calendar;

import java.util.Date;
import org.fireflow.engine.IRuntimeContextAware;
import org.fireflow.model.Duration;


/**
 *
 * @author chennieyun
 */
public interface ICalendarService extends IRuntimeContextAware{

    /**
     * Get the date after the duration
     * @param duration
     * @return
     */
    public Date dateAfter(Date fromDate, Duration duration);

    /**
     * 判断给定的日期是否为工作日
     * @param d
     * @return
     */
    public boolean isBusinessDay(Date d);

    public Date getSysDate();
}

File: trunk/dot-net_version/FireWorkflow.Net/FireWorkflow.Net/Engine/Calendar/ICalendarService.cs
using System;
using System.Collections.Generic;
using System.Text;
using FireWorkflow.Net.Model;

namespace FireWorkflow.Net.Engine.Calendar
{
    /// <summary>
    /// 日历服务
    /// </summary>
    public interface ICalendarService : IRuntimeContextAware
    {

        /// <summary>
        /// 获得fromDate后相隔duration的某个日期
        /// Get the date after the duration
        /// </summary>
        /// <param name="fromDate"></param>
        /// <param name="duration"></param>
        /// <returns></returns>
        DateTime dateAfter(DateTime fromDate, Duration duration);

        /// <summary>
        /// <para>缺省实现，周六周日都是非工作日，其他的都为工作日。</para>
        /// <para>实际应用中，可以在数据库中建立一张非工作日表，将周末以及法定节假日录入其中，</para>
        /// <para>然后在该方法中读该表的数据来判断工作日和非工作日。</para>
        /// </summary>
        /// <param name="d"></param>
        /// <returns></returns>
        Boolean isBusinessDay(DateTime d);

        /// <summary>
        /// 获得系统时间
        /// </summary>
        /// <returns></returns>
        DateTime getSysDate();
    }
}
